### PR TITLE
Remove the borrow of the device id string

### PR DESCRIPTION
**Pull request by basusamit**
While the borrow of the device id string minimizes the amount out copying, it does make the IoTHubClient harder to use.  The reference means you cannot, e.g., pass the client through a channel, or put it as a member of a struct due to the self-referential struct problem.  I don't see a significant downside of just copying the device ID into the client (I didn't actually see it being used, but I didn't want to just remove it).  If you prefer, I can try and create a "IoTHubClientOwned" version of the struct that owns the deviceID string.  There may be great reasons for the reference (and the associated lifetime) that I just don't understand.

### CODE REVIEW

**Comment by damienpontifex** [comment]
I'd been thinking the same thing 👍

File: src/client.rs [hunk 1]
```diff
@@ -57,9 +57,9 @@ where
     /// ```
     pub async fn new<TS>(
         hub_name: &str,
-        device_id: &'a str,
+        device_id: &str,
```
Comment (damienpontifex) [inline src/client.rs:60]: Could this take in `String` to let the user decide if they want to copy or transfer ownership?

File: src/client.rs [hunk 2]
```diff
@@ -115,10 +115,10 @@ where
     /// ```
     pub async fn new_with_transport(
         transport: TR,
-        device_id: &'a str,
-    ) -> crate::Result<IoTHubClient<'a, TR>> {
+        device_id: &str,
```
Comment (damienpontifex) [inline src/client.rs:118]: Same here with `String`?

File: Cargo.toml [hunk 3]
```diff
@@ -47,6 +47,7 @@ config = "0.10"
 dotenv = "0.15"
 serde_json = "1.0"
 chrono = { version = "0.4", features = ["serde"] }
+tokio-compat-02 = "0.1.2"
```
Comment (damienpontifex) [inline Cargo.toml:50]: Why was this needed?

Comment (basusamit) [reply Cargo.toml:50]: It's needed because the device-provisioning-service REST calls use hyper.  Hyper is not Tokio 0.3 compliant.  So if you try to make the REST calls under a 0.3 runtime, you just get a cryptic error message.  I updated the example to include the compat call to try and save people the headache.  But that meant it became a dev-dependency.

Comment (damienpontifex) [reply Cargo.toml:50]: Should it be optional then with changes becoming?
`tokio-compat-02 = { version = "0.1.2" optional = true }` 
then and features in Cargo.tomlL19-20 become
```
http-transport = ["hyper", "hyper-tls", "tokio-compat-02"]
with-provision = ["hyper", "hyper-tls", "tokio-compat-02"]
```

Comment (basusamit) [reply Cargo.toml]: Beyond that, I also moved the .compat call into the site where we need it (the various calls to hyper).  That makes it work out of the box.
